core/src/oauth/store.rs

```diff
@@ -219,12 +228,85 @@ impl OAuthStore for PostgresOAuthStore {
         Ok(())
     }
 
+    async fn create_credential(
+        &self,
+        provider: CredentialProvider,
+        credentials: serde_json::Map<String, serde_json::Value>,
+    ) -> Result<Credential> {
+        let pool = self.pool.clone();
+        let c = pool.get().await?;
+
+        let created = utils::now();
+        let secret = utils::new_id();
+
+        // Encrypt in database
+        let encrypted_credentials = Some(seal_str(&serde_json::to_string(&credentials)?)?);
```

Comment:
sealing/unsealing should be done outside of the store.rs file (if only out of consistency with the Connection code pathes + better division of concern)